test/modules/load-gen/Settings.cs

```diff
@@ -56,9 +60,27 @@ static Settings Create()
                 ? null
                 : configuration.GetValue<string>("testResultCoordinatorUrl");
 
-            string priorities = string.IsNullOrWhiteSpace(configuration.GetValue<string>("priorities"))
+            List<int> priorities = string.IsNullOrWhiteSpace(configuration.GetValue<string>("priorities"))
                 ? null
-                : configuration.GetValue<string>("priorities");
+                : configuration.GetValue<string>("priorities").Split(';').Select(int.Parse).ToList();
+
+            List<int> ttls = string.IsNullOrWhiteSpace(configuration.GetValue<string>("ttls"))
+                ? null
+                : configuration.GetValue<string>("ttls").Split(';').Select(int.Parse).ToList();
+
+            int ttlThresholdValue = configuration.GetValue<int>("ttlThresholdSecs");
+
+            if (!(ttls == null && priorities == null && ttlThresholdValue <= 0)
+                && !(ttls != null && priorities != null && ttlThresholdValue > 0))
+            {
+                throw new ArgumentException("TTLs, priorities, and TTLThreshold must either all be set or all not be set.");
+            }
+            else if (ttls.Count != priorities.Count && ttls.Count != 0)
```

Comment:
if ttls is null, priorities is null and ttlThresholdValue is 0, it will evaluate this line.  And it will throw NullReferenceException when calls ttls.Count.
```suggestion
            else if (ttls?.Count != 0 && ttls?.Count != priorities?.Count)
```

Reply:
Good catch